### PR TITLE
HV-1412 Take into account value unwrapping API changes

### PR DESCRIPTION
**Pull request by gsmet**
* https://hibernate.atlassian.net/browse/HV-1412

### CODE REVIEW

File: engine/src/main/java/org/hibernate/validator/metadata/descriptor/HibernateConstraintDescriptor.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import javax.validation.metadata.ConstraintDescriptor;
+
+/**
+ * Extension to {@code ConstraintDescriptor} which provides functionality
+ * specific to Hibernate Validator.
+ *
+ * @author Guillaume Smet
+ * @since 6.0
+ */
+public interface HibernateConstraintDescriptor<T extends Annotation> extends ConstraintDescriptor<T> {
```
Comment (gunnarmorling) [inline engine/src/main/java/org/hibernate/validator/metadata/descriptor/HibernateConstraintDescriptor.java]: What's the motivation for adding it already? Is it just for the sake of testing? I'd rather only add it if we actually need it.

Comment (gsmet) [inline engine/src/main/java/org/hibernate/validator/metadata/descriptor/HibernateConstraintDescriptor.java]: No, it's not only for testing. We also need it for the implementation as we need a reference type and I don't think using the `javax.validation` one makes sense.

I don't think it costs us much to have it prepared.

Comment (gunnarmorling) [inline engine/src/main/java/org/hibernate/validator/metadata/descriptor/HibernateConstraintDescriptor.java]: As discussed on HipChat I'm not in favor of adding it prematurely. People will only will wonder what the empty interface is about. For the `unwrap()` implementation, we can simply raise an exception at this point, as no type to unwrap to currently is supported. The alternative is to allow to unwrap into `ConstraintDescriptor` and super-types. I don't mind either way.

Comment (gsmet) [inline engine/src/main/java/org/hibernate/validator/metadata/descriptor/HibernateConstraintDescriptor.java]: OK, force pushed an update.

**Comment by gunnarmorling** [comment]
Thx!

**Pull request merged**